tests/lms/views/error_test.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from pyramid import httpexceptions
+import pytest
+
+from lms.views import error
+
+
+class TestErrorView:
+    def test_it_sets_correct_message_for_http_error(self, pyramid_request):
+        exc = httpexceptions.HTTPError('test http error msg', status_code=500)
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        resp = error_view.httperror()
+
+        assert resp['message'] == 'test http error msg'
+
+    def test_it_sets_correct_request_status_int_for_http_error(self, pyramid_request):
+        exc = httpexceptions.HTTPError('test http error msg', status_code=500)
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        error_view.httperror()
+
+        assert pyramid_request.response.status_int == exc.status_int
+
+    def test_it_sets_correct_message_for_http_server_error(self, pyramid_request):
+        exc = httpexceptions.HTTPServerError('test server error msg', status_code=500)
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        resp = error_view.httperror()
+
+        assert resp['message'] == 'test server error msg'
+
+    def test_it_sets_correct_request_status_int_for_http_server_error(self, pyramid_request):
+        exc = httpexceptions.HTTPServerError('test server error msg', status_code=500)
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        error_view.httperror()
+
+        assert pyramid_request.response.status_int == exc.status_int
+
+    def test_it_sets_correct_message_for_non_http_error(self, pyramid_request):
+        exc = Exception()
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        resp = error_view.error()
+
+        assert resp['message'] == 'Sorry, but something went wrong. The issue has been reported and we\'ll try to fix it.'
+
+    def test_it_logs_non_http_error_in_sentry(self, pyramid_request):
+        exc = Exception()
+
+        error_view = error.ErrorView(exc, pyramid_request)
+        error_view.error()
+
+        pyramid_request.raven.captureException.assert_called_once()
```

Comment:
This test will always pass, even if `captureException()` isn't called. The reason is that [the old third-party `mock` library that we're using has no `assert_called_once()` method](http://www.voidspace.org.uk/python/mock/search.html?q=assert_called_once&check_keywords=yes&area=default), and [`captureException()` is an unconstrained mock](https://github.com/hypothesis/lms/blob/master/tests/lms/conftest.py#L112). So `pyramid_request.raven.captureException.assert_called_once()` is the same as doing `pyramid_request.raven.captureException.foobar()` or `pyramid_request.raven.captureException.any_method_that_doesnt_exist()` - you can just call any non-existent method on this mock object and it will just succeed and return another mock object.

For this PR I think you should just change it to `assert pyramid_request.raven.captureException.call_count == 1`, but can you open two issues for future PRs:

1. We should change to using the new `unittest.mock` lib instead of the old `mock`
2. The `captureException` mock should use autospeccing